backend/api/Controllers/DocumentController.cs

```diff
@@ -54,38 +63,77 @@ public IActionResult GetDocumentTypes()
         [HttpGet]
         [HasPermission(Permissions.PropertyAdd)]
         [Produces("application/json")]
-        [ProducesResponseType(typeof(string), 200)]
+        [ProducesResponseType(typeof(ExternalResult<QueryResult<DocumentDetail>>), 200)]
         [SwaggerOperation(Tags = new[] { "documents" })]
         public IActionResult GetDocumentList()
         {
-            var ast = _documentService.GetDocumentList();
-            return new JsonResult(ast);
+            var result = _documentService.GetDocumentList();
+            return new JsonResult(result);
         }
 
         /// <summary>
         /// Downloads the file for the correspoding file and document id.
         /// </summary>
         [HttpGet("{documentId}/files/{fileId}/download")]
         [HasPermission(Permissions.PropertyAdd)]
-        [ProducesResponseType(typeof(string), 200)]
+        [ProducesResponseType(typeof(ExternalResult<FileDownload>), 200)]
         [SwaggerOperation(Tags = new[] { "documents" })]
         public IActionResult DownloadFile(int documentId, int fileId)
         {
-            var ast = _documentService.DownloadFile(documentId, fileId);
-            return new JsonResult(ast);
+            var result = _documentService.DownloadFile(documentId, fileId);
+            return new JsonResult(result);
         }
 
         /// <summary>
         /// Uploads the passed document.
         /// </summary>
         [HttpPost]
         [HasPermission(Permissions.PropertyAdd)]
-        [ProducesResponseType(typeof(string), 200)]
+        [ProducesResponseType(typeof(ExternalResult<DocumentDetail>), 200)]
         [SwaggerOperation(Tags = new[] { "documents" })]
         public IActionResult UploadDocument([FromForm] int documentType, [FromForm] IFormFile file)
         {
-            var ast = _documentService.UploadDocument(documentType, file);
-            return new JsonResult(ast);
+            var result = _documentService.UploadDocument(documentType, file);
+            return new JsonResult(result);
+        }
+
+        /// <summary>
+        /// Uploads the passed document.
+        /// </summary>
+        [HttpPatch("sync/mayan/documenttype")]
+        //[HasPermission(Permissions.PropertyAdd)]
+        [ProducesResponseType(typeof(ExternalBatchResult), 200)]
+        [SwaggerOperation(Tags = new[] { "documents" })]
+        public IActionResult SyncMayanDocumentTypes([FromBody] SyncModel model)
+        {
+            var result = _documentSyncService.SyncMayanDocumentTypes(model);
+            return new JsonResult(result);
+        }
+
+        /// <summary>
+        /// Uploads the passed document.
+        /// </summary>
+        [HttpPatch("sync/mayan/metadatatype")]
+        //[HasPermission(Permissions.PropertyAdd)]
```

Comment:
nit: putting a TODO here might help come back to this later when we have the appropriate permissions set up

Reply:
thats a good idea